geode-apis-compatible-with-redis/src/integrationTest/java/org/apache/geode/redis/internal/executor/pubsub/AbstractSubCommandsIntegrationTest.java

```diff
@@ -191,6 +201,67 @@ public void channels_shouldNotReturnDuplicates_givenMultipleSubscribersToSameCha
     subscriber2.close();
   }
 
+  /** -- NUMSUB-- **/
+
+  @Test
+  public void numsub_shouldReturnEmptyList_whenCalledWithOutChannelNames() {
+    executor.submit(() -> subscriber.subscribe(mockSubscriber, "foo"));
+    waitFor(() -> mockSubscriber.getSubscribedChannels() == 1);
+
+    List<Object> result =
+        uncheckedCast(introspector.sendCommand(Protocol.Command.PUBSUB, PUBSUB_NUMSUB));
+
+    assertThat(result).isEmpty();
+  }
+
+  @Test
+  public void numsub_shouldReturnListOfChannelsWithSubscriberCount_whenCalledWithActiveChannels() {
+    Jedis subscriber2 = new Jedis(BIND_ADDRESS, getPort(), REDIS_CLIENT_TIMEOUT);
+    MockSubscriber fooAndBarSubscriber = new MockSubscriber();
+
+    executor.submit(() -> subscriber.subscribe(mockSubscriber, "foo"));
+    executor.submit(() -> subscriber2.subscribe(fooAndBarSubscriber, "foo", "bar"));
+    waitFor(() -> mockSubscriber.getSubscribedChannels() == 1
+        && fooAndBarSubscriber.getSubscribedChannels() == 2);
+
+    HashMap<String, String> result =
```

Comment:
If you use `Map<String, String>` here, you can remove the cast.

Reply:
Done